builder.go

```diff
@@ -563,14 +563,21 @@ var builtinAllowedBuildArgs = map[string]bool{
 }
 
 // ParseDockerIgnore returns a list of the excludes in the .dockerignore file.
-// extracted from fsouza/go-dockerclient.
+// extracted from fsouza/go-dockerclient and modified to drop comments and
+// empty lines.
 func ParseDockerignore(root string) ([]string, error) {
 	var excludes []string
 	ignore, err := ioutil.ReadFile(filepath.Join(root, ".dockerignore"))
 	if err != nil && !os.IsNotExist(err) {
 		return excludes, fmt.Errorf("error reading .dockerignore: '%s'", err)
 	}
-	return strings.Split(string(ignore), "\n"), nil
+	for _, e := range strings.Split(string(ignore), "\n") {
+		if len(e) == 0 || e[0] == '#' {
```

Comment:
should this be
```suggestion
		if len(e) == 0 || strings.HasPrefix(e[0] == '#') {
```
or do we process commented lines if they're not empty?

Reply:
This should be skipping all empty or commented lines (well, commented in the first column, per docs).  Are you seeing it missing a case that it should be catching?

Reply:
I may have the logic reveresed here, but here's what I was looking at.  Given a string `# Test`, I expected that line to be ignored (hit the continue at line 576).   When I ran a small piece of go code,  the if statement in line 575 did not equate to 'True` for that line as `e[0]` contained `# Test` and not `#`.  I was thinking it should fire for that line, or have I been playing too much reversi lately?

Reply:
`strings.Split()` should return a slice of strings, so if we're iterating through that, `e` will be a string, and its elements will be runes.

Reply:
@nalind Thanks, that was my bad, I was working on an array of strings in my tests.  Reworked my test and it fires as you intended.